src/Jaeger.Core/Metrics/MetricsImpl.cs

```diff
@@ -0,0 +1,173 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+namespace Jaeger.Core.Metrics
+{
+    public class MetricsImpl : IMetrics
+    {
+        /// <inheritdoc/>
+        [Metric(name: "traces", tag1Key: "state", tag1Value: "started", tag2Key: "sampled", tag2Value: "y")]
```

Comment:
Is this a limitation of the annotation syntax that you can't do a dictionary?

If so I would recommend just having a string like `tags="state=started, sampled=y"` and parse it during construction. The objective is to make the annotation readable & easy, not to help the parsing engine (which isn't that complicated anyway).

Reply:
Yes, this is a limitation in C#! I'll change it to your recommendation.